events/UserAllocatedEvent.h

```diff
@@ -260,17 +262,22 @@ class UserAllocatedEvent<F, void(ArgTs...)> {
             return false;
         }
         core_util_atomic_incr_u8(&_post_ref, 1);
+        equeue_event_delay(&_e + 1, _delay);
```

Comment:
I think that changes the behavior? As before we could change delay and period of already posted events?

Reply:
It wasn't allowed previously. It was guarded by  `MBED_ASSERT(!_post_ref)`.
It isn't a valid use case to change delay/period while dispatching is ongoing.